test/src/api/responder04Test.php

```diff
@@ -0,0 +1,916 @@
+        $expData['playerDataArray'][0]['activeDieArray'][2]['description'] = "Poison Turbo V Swing Die (with 10 sides)";
+        $expData['playerDataArray'][0]['activeDieArray'][2]['sides'] = 10;
+        $expData['playerDataArray'][0]['activeDieArray'][2]['value'] = 10;
+        $expData['playerDataArray'][0]['activeDieArray'][3]['description'] = "Null Turbo Twin Y Swing Die (both with 11 sides)";
+        $expData['playerDataArray'][0]['activeDieArray'][3]['sides'] = 22;
+        $expData['playerDataArray'][0]['activeDieArray'][3]['subdieArray'] = array(array("sides" => "11", "value" => "11"), array("sides" => "11", "value" => "6"));
+        $expData['playerDataArray'][0]['activeDieArray'][3]['value'] = 17;
+        $expData['playerDataArray'][0]['activeDieArray'][4]['value'] = 5;
+        $expData['playerDataArray'][0]['activeDieArray'][4]['wildcardPropsArray'] = array("colour" => "red", "displayedValue" => '5<span class="suit_red">&diams;</span>', "suit" => "Diamonds", "type" => "Wildcard");
+        $expData['playerDataArray'][0]['roundScore'] = 9;
+        $expData['playerDataArray'][0]['sideScore'] = -9.7;
+        $expData['playerDataArray'][0]['swingRequestArray'] = array();
+        $expData['playerDataArray'][0]['turboSizeArray'] = array("2" => array(6, 7, 8, 9, 10, 11, 12), "3" => array(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20));
+        $expData['playerDataArray'][0]['waitingOnAction'] = false;
+        $expData['playerDataArray'][1]['activeDieArray'][0]['value'] = 4;
+        $expData['playerDataArray'][1]['activeDieArray'][1]['value'] = 4;
+        $expData['playerDataArray'][1]['activeDieArray'][2]['value'] = 1;
+        $expData['playerDataArray'][1]['activeDieArray'][3]['value'] = 1;
+        $expData['playerDataArray'][1]['activeDieArray'][4]['description'] = "Doppelganger X Swing Die (with 15 sides)";
+        $expData['playerDataArray'][1]['activeDieArray'][4]['sides'] = 15;
+        $expData['playerDataArray'][1]['activeDieArray'][4]['value'] = 13;
+        $expData['playerDataArray'][1]['roundScore'] = 23.5;
+        $expData['playerDataArray'][1]['sideScore'] = 9.7;
+        $expData['playerDataArray'][1]['swingRequestArray'] = array();
+        $expData['playerDataArray'][1]['waitingOnAction'] = true;
+        $expData['playerWithInitiativeIdx'] = 1;
+        $expData['validAttackTypeArray'] = array("Power", "Skill");
+
+        $retval = $this->verify_api_loadGameData($expData, $gameId, 10);
+
+        $_SESSION = $this->mock_test_user_login('responder004');
+        $this->verify_api_submitTurn(
+            array(7),
+            'responder004 performed Power attack using [D(X=15):13] against [p(V=10)!:10]; Defender p(V=10)! was captured; Attacker D(X=15) changed size from 15 to 10 sides, recipe changed from D(X=15) to p(V=10)!, rerolled 13 => 7. ',
+            $retval, array(array(1, 4), array(0, 2)),
+            $gameId, 1, 'Power', 1, 0, '', array());
```

Comment:
Here's an example of what i believe is supposed to happen when you doppelganger a turbo die --- no turbo values are sent, no turbo change happens, we move on to the next attack.